src/tiled/terrainbrush.cpp

```diff
@@ -371,6 +372,34 @@ void TerrainBrush::updateBrush(QPoint cursorPos, const QVector<QPoint> *list)
         int x = p.x(), y = p.y();
         int i = y*layerWidth + x;
 
+        // to support isometric staggered, make edges into variables
+        int upperID = i - layerWidth;
+        int bottomID = i + layerWidth;
+        int leftID = i - 1;
+        int rightID = i + 1;
+
+        QPoint upPoint(p.x(),p.y()-1);
+        QPoint bottomPoint(p.x(),p.y()+1);
+        QPoint leftPoint(p.x()-1,p.y()-1);
+        QPoint rightPoint(p.x()+1,p.y());
+
+        if(mapDocument() && mapDocument()->map() && mapDocument()->map()->orientation() == Map::Staggered){
+            StaggeredRenderer* renderer = dynamic_cast<StaggeredRenderer*>(mapDocument()->renderer());
+            if(renderer){
+                upPoint = renderer->topRight(p.x(),p.y());
+                upperID = upPoint.y()*layerWidth + upPoint.x();
+
+                bottomPoint = renderer->bottomLeft(p.x(),p.y());
+                bottomID = bottomPoint.y()*layerWidth + bottomPoint.x();
+
+                leftPoint = renderer->topLeft(p.x(),p.y());
+                leftID = leftPoint.y()*layerWidth + leftPoint.x();
+
+                rightPoint = renderer->bottomRight(p.x(),p.y());
+                rightID = rightPoint.y()*layerWidth + rightPoint.x();
```

Comment:
You can move the assignments to `rightID` etc. out of the condition, since the same calculation works for orthogonal maps. 4 lines less code. :-)